langchain/requests.py

```diff
@@ -3,71 +3,131 @@
+        self, url: str, data: Dict[str, Any], **kwargs: Any
+    ) -> aiohttp.ClientResponse:
         """PUT the URL and return the text asynchronously."""
         return await self._arequest("PUT", url, json=data, **kwargs)
 
-    async def adelete(self, url: str, **kwargs: Any) -> str:
+    async def adelete(self, url: str, **kwargs: Any) -> aiohttp.ClientResponse:
         """DELETE the URL and return the text asynchronously."""
         return await self._arequest("DELETE", url, **kwargs)
+
+
+class TextRequestsWrapper(BaseModel):
+    """Lightweight wrapper around requests library."""
+
+    headers: Optional[Dict[str, str]] = None
+    aiosession: Optional[aiohttp.ClientSession] = None
+
+    @property
+    def requests(self) -> Requests:
+        return Requests(headers=self.headers, aiosession=self.aiosession)
+
+    def get(self, url: str, **kwargs: Any) -> str:
+        """GET the URL and return the text."""
+        return self.requests.get(url, **kwargs).text
+
+    def post(self, url: str, data: Dict[str, Any], **kwargs: Any) -> str:
+        """POST to the URL and return the text."""
+        return self.requests.post(url, json=data, headers=self.headers, **kwargs).text
+
+    def patch(self, url: str, data: Dict[str, Any], **kwargs: Any) -> str:
+        """PATCH the URL and return the text."""
+        return self.requests.patch(url, json=data, headers=self.headers, **kwargs).text
+
+    def put(self, url: str, data: Dict[str, Any], **kwargs: Any) -> str:
+        """PUT the URL and return the text."""
+        return self.requests.put(url, json=data, headers=self.headers, **kwargs).text
+
+    def delete(self, url: str, **kwargs: Any) -> str:
+        """DELETE the URL and return the text."""
+        return self.requests.delete(url, headers=self.headers, **kwargs).text
+
+    async def aget(self, url: str, **kwargs: Any) -> str:
+        """GET the URL and return the text asynchronously."""
+        response = await self.requests.aget(url, **kwargs)
+        return await response.text()
+
+    async def apost(self, url: str, data: Dict[str, Any], **kwargs: Any) -> str:
+        """POST to the URL and return the text asynchronously."""
+        response = await self.requests.apost(url, data, **kwargs)
+        return await response.text()
+
+    async def apatch(self, url: str, data: Dict[str, Any], **kwargs: Any) -> str:
+        """PATCH the URL and return the text asynchronously."""
+        response = await self.requests.apatch(url, data, **kwargs)
+        return await response.text()
+
+    async def aput(self, url: str, data: Dict[str, Any], **kwargs: Any) -> str:
+        """PUT the URL and return the text asynchronously."""
+        response = await self.requests.aput(url, data, **kwargs)
+        return await response.text()
+
+    async def adelete(self, url: str, **kwargs: Any) -> str:
+        """DELETE the URL and return the text asynchronously."""
+        response = await self.requests.aget(url, **kwargs)
```

Comment:
Should this be `self.requests.adelete(url, **kwargs)` instead of `self.requests.aget(url, **kwargs)`?

Reply:
yes good catch, thanks